internal/dataplane/parser/parser.go

```diff
@@ -39,6 +39,41 @@ const (
 // Parser - Public Types
 // -----------------------------------------------------------------------------
 
+type ParsingError struct {
+	relatedObjects []client.Object
+	reason         string
+}
+
+func NewParsingError(reason string, relatedObjects ...client.Object) ParsingError {
+	if reason == "" {
+		reason = "unknown"
+	}
+	return ParsingError{
+		relatedObjects: relatedObjects,
+		reason:         reason,
+	}
+}
+
+func (p ParsingError) RelatedObjects() []client.Object {
+	return p.relatedObjects
+}
+
+func (p ParsingError) Reason() string {
+	return p.reason
+}
+
+type parsingErrorsCollector struct {
+	errors []ParsingError
+}
+
+func newParsingErrorsCollector() *parsingErrorsCollector {
+	return &parsingErrorsCollector{}
+}
+
+func (c *parsingErrorsCollector) ParsingError(reason string, relatedObjects ...client.Object) {
+	c.errors = append(c.errors, NewParsingError(reason, relatedObjects...))
+}
```

Comment:
`ParsingError` is used to gather parsing errors during the translation. It accepts related/affected objects regarding the error.